### ISSUE TITLE
Title of WMTS query results

### ISSUE THREAD

**Issue by rbovard** [camptocamp/ngeo#2604]
Project:

* GeoMapFish version: `2.1.4`

[Permalink directly where we can reproduce the issue](https://map-demo.nyon.ch/v2/mobile/theme/cadastre?lang=fr&baselayer_ref=plan_cadastral&tree_groups=adresses%2Ccadastre%2Climites_administratives&tree_group_layers_adresses&tree_group_layers_cadastre&tree_group_layers_limites_administratives&map_x=507843&map_y=137446&map_zoom=10)

Steps to reproduce the issue:

* Query the building **666**

Result:
* The title of the 2 results are "Plan cadastral"

Expected result:
* The title of the 2 results should be "Parcelles" and "Bâtiments" like in [desktop](https://map-demo.nyon.ch/v2/theme/cadastre?map_x=507843&map_y=137446&map_zoom=10&tree_group_layers_adresses=&tree_group_layers_cadastre=&tree_group_layers_limites_administratives=)

**Comment by fredj** [comment]
But the results are really coming from the "Plan cadastral" layer (the background layer)

**Comment by fredj** [comment]
ok, I see now that in the desktop interface the background layer is also queryable

**Comment by rbovard** [comment]
Nope... As you can see in desktop.

And for the mobile:

```
<gml:featureMember>
  <ms:mo_par_prive>
...

<gml:featureMember>
  <ms:mo_baths>
...
```
Where `mo_par_prive` is "Parcelles" and `mo_baths` is "Bâtiments

**Comment by fredj** [comment]
@rbovard the "Plan cadastral" background layer is not available anymore on https://map-demo.nyon.ch/v2/mobile/

**Comment by rbovard** [comment]
@fredj Yes, it's due to https://github.com/camptocamp/c2cgeoportal/issues/2934

I'm waiting on next release...

**Comment by ybolognini** [comment]
@rbovard Did you update c2cgeoportal? Is this fixed?

**Comment by rbovard** [comment]
Not yet, I'm coming back from holidays today.

I'll keep you posted since it's tested - as all my opened issues like usual - but not before next week.

**Comment by sbrunner** [comment]
What's the status of this issue?

**Comment by rbovard** [comment]
@fredj  I just tested in `2.1.8` and it's not working. Could you please re open this issue?

FYI the original permalink is still valid to reproduce it.

**Comment by ybolognini** [comment]
@rbovard 
Query features have been refactored for the 2.2. Do you plan to switch to 2.2, now that it is released?

**Comment by ger-benjamin** [comment]
I've investigated this issue. It looks like it's a problem from Openlayers. Currently we cannot know which layer is linked with which returned layers (this information is in the sever's response but not added in the feature by Openlayers.) We will investigate what we can do inside Openlayers then we come back on this issue.

**Comment by fredj** [comment]
We have a WFS query on a source which returns multiple features.
What is displayed as the title in the result window is the name of the query source (in this case "Plan cadastral").

**Comment by fredj** [comment]
OpenLayers doesn't return the type of the feature (GML parser). It was the case with OpenLayers 2

**Comment by fredj** [comment]
Can be tested with:
https://geomapfish-demo.camptocamp.net/2.2/theme/Demo?lang=en&baselayer_ref=OSM%20map&tree_group_opacity_Group=0.65&tree_enable_osm_scale=true&tree_group_layers_Layers=cinema%2Cpost_office%2Centertainment%2Csustenance%2Chospitals&tree_group_layers_Group=osm&tree_group_layers_OSM%20functions=osm_scale&tree_group_layers_External&tree_enable_osm_open&tree_enable_ch.swisstopo.geologie-geotechnik-gk500-gesteinsklassierung&tree_enable_OSM%2520map&tree_enable_osm_time_r_dp_2&tree_group_layers_Filters&map_x=616289&map_y=176702&map_zoom=4&tree_groups=Group

This group has one children `osm` and three `childLayers`: `hotel`, `bank` and `place_of_worship`.
Only one datasource is create for `osm`.

**Comment by rbovard** [comment]
@fredj Are you planning to report this issue to OpenLayers?

**Comment by fredj** [comment]
The problem is in ngeo; we're working on it

**Comment by rbovard** [comment]
Thanks now the title is correct but **not translated**... Same steps as in first message with this [permalink](https://map-demo.nyon.ch/dev/mobile/theme/cadastre?lang=fr&baselayer_ref=plan_cadastral&tree_groups=adresses%2Ccadastre%2Climites_administratives&tree_group_layers_adresses&tree_group_layers_cadastre&tree_group_layers_limites_administratives&map_x=507843&map_y=137446&map_zoom=10).

@fredj Should I open a new issue?
